src/ConnectedMode/UI/Credentials/CredentialsViewModel.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.ObjectModel;
+using System.IO;
+using SonarLint.VisualStudio.ConnectedMode.UI.Resources;
+using SonarLint.VisualStudio.Core.WPF;
+using static SonarLint.VisualStudio.ConnectedMode.ConnectionInfo;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UI.Credentials;
+
+public class CredentialsViewModel(Connection connection) : ViewModelBase
+{
+    private string token;
+    private string selectedAuthenticationType = UiResources.Token;
+    private string username;
+    private string password;
+
+    public Connection Connection { get; } = connection;
+    public ObservableCollection<string> AuthenticationType { get; } = [UiResources.Token, UiResources.Credentials];
+
+    public string SelectedAuthenticationType
+    {
+        get => selectedAuthenticationType;
+        set
+        {
+            selectedAuthenticationType = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsTokenAuthentication));
+            RaisePropertyChanged(nameof(IsCredentialsAuthentication));
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldTokenBeFilled));
+            RaisePropertyChanged(nameof(ShouldUsernameBeFilled));
+            RaisePropertyChanged(nameof(ShouldPasswordBeFilled));
+        }
+    }
+
+    public string Token
+    {
+        get => token;
+        set
+        {
+            token = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldTokenBeFilled));
+        }
+    }
+
+    public string Username
+    {
+        get => username;
+        set
+        {
+            username = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
```

Comment:
What do you think about having a full property instead of the computed one for `IsConfirmationBtnEnabled` , setting the value here and updating `RaisePropertyChanged` in the `IsConfirmationBtnEnabled` setter? That would remove the need to keep track of all of the property changed and delegate that.

Reply:
I wouldn't do that, because I think it would violate the single responsibility principle. Also the IsConfirmationBtnEnabled needs to be set in all those places, where we trigger the RaisePropertyChanged: when the username changes, the password, token etc